pulumi/grapl/__main__.py

```diff
@@ -41,24 +42,37 @@ def _get_subset(inputs: NomadVars, subset: Set[str]) -> NomadVars:
     return {k: inputs[k] for k in subset}
 
 
-def grapl_core_docker_image_tags(
+def _container_images(
     artifacts: Mapping[str, str], require_artifact: bool = False
-) -> NomadVars:
-    # partial apply some repeated args
-    version_tag_alias = lambda key: version_tag(key, artifacts, require_artifact)
-
-    return dict(
-        analyzer_dispatcher_tag=version_tag_alias("analyzer-dispatcher"),
-        analyzer_executor_tag=version_tag_alias("analyzer-executor"),
-        dgraph_tag="latest",
-        engagement_creator_tag=version_tag_alias("engagement-creator"),
-        graph_merger_tag=version_tag_alias("graph-merger"),
-        graphql_endpoint_tag=version_tag_alias("graphql-endpoint"),
-        model_plugin_deployer_tag=version_tag_alias("model-plugin-deployer"),
-        node_identifier_tag=version_tag_alias("node-identifier"),
-        osquery_generator_tag=version_tag_alias("osquery-generator"),
-        sysmon_generator_tag=version_tag_alias("sysmon-generator"),
-        web_ui_tag=version_tag_alias("grapl-web-ui"),
+) -> str:
+    """
+    Build a map of "task name -> docker image URL".
+    See the .nomad file's var `container_images` for further documentation.
+    """
+    img_id_builder = DockerImageIdBuilder(config.container_repository())
+
+    def build_img_id_with_tag(image_name: str) -> str:
+        # A shortcut to grab the version tag from the artifacts map and build a
+        # Cloudsmith docker image url out of it.
+        tag = version_tag(image_name, artifacts, require_artifact)
+        return img_id_builder.build(image_name=image_name, tag=tag)
+
+    return json.dumps(
```

Comment:
I think it'd be a bit more generic and decoupled to have this function return the `Mapping[str, str]`, and convert to a JSON string at the point of consumption that requires it to actually be a string.

Reply:
I've moved this logic to NomadJob; honestly this should be handled for us by the nomad-pulumi plugin, but the NomadJob abstraction is a good place to handle it.